packages/shared-core/src/table.ts

```diff
@@ -51,3 +52,22 @@ export function canBeDisplayColumn(type: FieldType): boolean {
 export function canBeSortColumn(type: FieldType): boolean {
   return !!allowSortColumnByType[type]
 }
+
+export function findDuplicateInternalColumns(table: Table): string | undefined {
```

Comment:
For the sake of a slightly better error message, is it worth here returning a list of all duplicated columns? So in the error we can give the user all the columns they need to pay attention to?